nsv13/code/modules/power/stormdrive.dm

```diff
@@ -1520,19 +1527,50 @@ Control Rods
 
 /obj/effect/landmark/nuclear_waste_spawner //Clean way of spawning nuclear gunk after a reactor core meltdown.
 	name = "nuclear waste spawner"
-	var/range = 5 //5 tile radius to spawn goop
+	var/range = PLUTONIUM_SLUDGE_RANGE //tile radius to spawn goop
+	var/list/avoid_objs = list( // List of objs that the waste does not spawn on
+		/obj/structure/stairs, // Sludge is hidden below stairs
+		/obj/structure/ladder, // Going down the ladder directly on sludge bad
+		/obj/effect/decal/nuclear_waste, // No stacked sludge
+		)
 
 /obj/effect/landmark/nuclear_waste_spawner/strong
-	range = 10
+	range = PLUTONIUM_SLUDGE_RANGE_STRONG
 
 /obj/effect/landmark/nuclear_waste_spawner/proc/fire()
 	playsound(loc, 'sound/effects/gib_step.ogg', 100)
-	new /obj/effect/decal/nuclear_waste/epicenter(get_turf(src))
+
+	place_sludge(get_turf(src), TRUE)
 	for(var/turf/open/floor in orange(range, get_turf(src)))
-		if(prob(35)) //Scatter the sludge, don't smear it everywhere
-			new /obj/effect/decal/nuclear_waste (floor)
+		place_sludge(floor, FALSE)
+
 	qdel(src)
 
+/// Tries to place plutonium sludge on 'floor'. Returns TRUE if the turf has been successfully processed, FALSE otherwise.
+/obj/effect/landmark/nuclear_waste_spawner/proc/place_sludge(turf/open/floor, epicenter = FALSE)
```

Comment:
Why do you have an extra proc here for inserting just into fire() anyway?

Reply:
I thought the code would be cleaner if the sludge placement code was separated to another proc instead of the main fire() proc. Also the 'orange' returns all turfs except the center so i elected to call my new proc twice, one for the center dense sludge and one normal sludge for every turf except the center.